src/main/java/com/microsoft/graph/content/BatchResponseContent.java

```diff
@@ -0,0 +1,228 @@
+        if (jsonBatchResponseObject != null) {
+            JsonElement responsesElement = jsonBatchResponseObject.get(CoreConstants.BatchRequest.RESPONSES);
+            if (responsesElement != null && responsesElement.isJsonArray()) { //ensure "responses" is not null and is an array.
+                JsonArray responsesArray = responsesElement.getAsJsonArray();
+                for (JsonElement responseElement : responsesArray) {
+                    if (responseElement.getAsJsonObject().get("id").getAsString().equals(requestId)) {
+                        return CompletableFuture.completedFuture(getResponseFromJsonObject(responseElement));
+                    }
+                }
+            }
+        }
+        return CompletableFuture.completedFuture(null);
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @param responseHandler The response handler to use when deserializing the response body.
+     * @return The response within the batch response via specified id, null if not found.
+     * @param <T> The type of the response body.
+     */
+    @Nullable
+    public <T extends Parsable> CompletableFuture<T> getResponseByIdAsync(@Nonnull String requestId, @Nonnull ResponseHandler responseHandler) {
+        Response response = getResponseByIdAsync(requestId).join();
+        if(response == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        return responseHandler.handleResponseAsync(response, apiErrorMappings);
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @param factory The factory to use when deserializing the response body.
+     * @return The response within the batch response via specified id, null if not found.
+     * @param <T> The type of the response body.
+     */
+    @Nullable
+    public <T extends Parsable> CompletableFuture<T> getResponseByIdAsync(@Nonnull String requestId, @Nonnull ParsableFactory<T> factory) {
+        return this.getResponseByIdAsync(requestId, new ResponseBodyHandler<>(factory));
+    }
+    /**
+     * Gets the response within the batch response via specified id.
+     * @param requestId The id of the request.
+     * @return The response within the batch response via specified id as an InputStream, null if not found.
+     */
+    @Nullable
+    @SuppressFBWarnings
+    public CompletableFuture<InputStream> getResponseStreamByIdAsync(@Nonnull String requestId) {
+        Response response = getResponseByIdAsync(requestId).join();
+        if(response != null && response.body() != null) {
+            try{
+                InputStream in = response.body().byteStream();
+                ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(ByteStreams.toByteArray(in));
```

Comment:
is there any specific reason why you're copying the stream here?

Reply:
This may have been for explicit testing, removing.